File: leetcode/python/268_missing_number.py
class Solution:
    def missingNumber(self, nums: list[int]) -> int:
        bucket = set(range(0, len(nums)+1))
        for n in nums:
            bucket.remove(n)
        return bucket.pop()

s = Solution()
print(s.missingNumber([3,0,1]))
print(s.missingNumber([0,1]))
print(s.missingNumber([9,6,4,2,3,5,7,0,1]))

File: leetcode/rust/src/268_missing_number.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let target = (1..nums.len() as i32 + 1).into_iter().sum::<i32>();
        let sum = nums.into_iter().sum::<i32>();
        target - sum
    }
}